test/crud-with-options.test.js

```diff
@@ -536,6 +536,54 @@ describe('crud-with-options', function() {
 
   });
 
+  describe('upsertWithWhere', function() {
+
+    beforeEach(seed);
+
+    it('should allow upsertWithWhere(data, cb)', function(done) {
```

Comment:
Actually, it's not a regular `upsert`, because the semantics is different. Regular `upsert` checks for `id`, while `upsertWithWhere` with no where works only as long as there is 0 or 1 instance (table row). I am concerned this would confuse LoopBack users and we will have to spend more time on support as a result.